src_c/draw.c

```diff
@@ -897,19 +906,44 @@ rect(PyObject *self, PyObject *args, PyObject *kwargs)
 
     if (radius <= 0 && top_left_radius <= 0 && top_right_radius <= 0 &&
         bottom_left_radius <= 0 && bottom_right_radius <= 0) {
-        l = rect->x;
-        r = rect->x + rect->w - 1;
-        t = rect->y;
-        b = rect->y + rect->h - 1;
-        points = Py_BuildValue("((ii)(ii)(ii)(ii))", l, t, r, t, r, b, l, b);
-        poly_args = Py_BuildValue("(OONi)", surfobj, colorobj, points, width);
-        if (NULL == poly_args) {
-            return NULL; /* Exception already set. */
+        if(width > 0){
+            l = rect->x;
+            r = rect->x + rect->w - 1;
+            t = rect->y;
+            b = rect->y + rect->h - 1;
+            points = Py_BuildValue("((ii)(ii)(ii)(ii))", l, t, r, t, r, b, l, b);
+            poly_args = Py_BuildValue("(OONi)", surfobj, colorobj, points, width);
+            if (NULL == poly_args) {
+                return NULL; /* Exception already set. */
+            }
+
+            ret = polygon(NULL, poly_args, NULL);
+            Py_DECREF(poly_args);
+            return ret;
         }
+        else {
+            sdlrect.x = rect->x;
+            sdlrect.y = rect->y;
+            sdlrect.w = rect->w;
+            sdlrect.h = rect->h;
 
-        ret = polygon(NULL, poly_args, NULL);
-        Py_DECREF(poly_args);
-        return ret;
+            SDL_GetClipRect(surf, &cliprect);
+
+
+            if (!SDL_IntersectRect(&sdlrect,
```

Comment:
Is the old draw.rect code returning the same rect as this one? We need a test case for this.

There is `test_rect__surface_clip` but it doesn't seem to assert against the returned rect.

Reply:
For that we have `test_rect__bounding_rect` function: https://github.com/pygame/pygame/blob/3ab109795a90489440856a269fd033545de6c694/test/draw_test.py#L4815

Reply:
Cool.

However, it doesn't use `Surface.set_clip`.